src/reconcile.py

```diff
@@ -0,0 +1,274 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for calculating required changes based on docs directory and navigation table."""
+
+import itertools
+import typing
+
+from . import exceptions, types_
+from .discourse import Discourse
+
+
+def _local_only(path_info: types_.PathInfo) -> types_.CreateAction:
+    """Return a create action based on information about a local documentation file.
+
+    Args:
+        path_info: Information about the local documentation file.
+
+    Returns:
+        A page create action.
+    """
+    return types_.CreateAction(
+        action=types_.Action.CREATE,
+        level=path_info.level,
+        path=path_info.table_path,
+        navlink_title=path_info.navlink_title,
+        content=path_info.local_path.read_text() if path_info.local_path.is_file() else None,
+    )
+
+
+def _local_and_server(
+    path_info: types_.PathInfo, table_row: types_.TableRow, discourse: Discourse
+) -> (types_.UpdateAction | types_.NoopAction | types_.CreateAction | types_.DeleteAction):
+    """Return an update or noop action depending on if the content or navlink title has changed.
+
+    Args:
+        path_info: Information about the local documentation file.
+        table_row: A row from the navigation table.
+        discourse: A client to the documentation server.
+
+    Returns:
+        A page update or noop action.
+
+    Raises:
+        ReconcilliationError: if the table path or level do not match for the path info and table
+            row.
```

Comment:
Well spotted, yes this was outdated. Writing it also made me realise there is a bug in the case where there is a directory locally and a page on the server, just deleting the page on the server will mean that the grouping is not added to the navlink. Need to think about how to handle that case since it currently would require 2 actions to be returned which would need to be flattened. I'm thinking I'll change the `run_one` function to always return multiple actions and add an unpack on `run_all`, that would be the simplest

Reply:
This has now been pushed to this PR

Reply:
Cool, indeed the fact that we were only deleting here was instinctively strange, but couldn't point out why. I'll review this change.